$NetBSD: patch-aa,v 1.3 2011/08/01 08:42:39 apb Exp $

Avoid an infinite loop in the configure script after 
"checking for replacing readdir using getdirentries()..."

--- ../lib/replace/test/os2_delete.c.orig	2011-07-24 20:04:44.000000000 +0000
+++ ../lib/replace/test/os2_delete.c	2011-08-01 08:09:59.000000000 +0000
@@ -116,6 +116,7 @@
 	while (1) {
 		int n = os2_delete(d);
 		if (n == 0) break;
+		if (test_readdir_os2_delete_ret != 0) break; /* if FAILED */
 		total_deleted += n;
 	}
 	closedir(d);
